File: src/main/java/ru/market/pricing_service/model/entity/ProductPriceInCheck.java
package ru.market.pricing_service.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.market.inventory_service.model.entity.Product;
import java.math.BigDecimal;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductPriceInCheck {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer productPriceInCheckId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "check_id", nullable = false)
    private Check check;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "price_list_id", nullable = false)
    private PriceList priceList;

    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "discount_id", nullable = false)
    private Discount discount;

    @Column(name = "input_price", precision = 10, scale = 2)
    private BigDecimal inputPrice;

    @Column(name = "final_price", precision = 10, scale = 2)
    private BigDecimal finalPrice;

    @Column(name = "price_type")
    private String priceType;
}

File: src/main/java/ru/market/pricing_service/model/entity/ProductPriceInCheck.kt
package ru.market.pricing_service.model.entity

import jakarta.persistence.*
import ru.market.inventory_service.model.entity.Product
import java.math.BigDecimal

@Entity
data class ProductPriceInCheck(
    @Id
    @GeneratedValue
    val productPriceInCheckId: Int,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    val product: Product,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "check_id", nullable = false)
    val check: Check,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "price_list_id", nullable = false)
    val priceList: PriceList,

    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "discount_id", nullable = false)
    val discount: Discount,

    @Column(precision = 10, scale = 2) val inputPrice: BigDecimal,
    @Column(precision = 10, scale = 2) val finalPrice: BigDecimal,
    @Column(precision = 10, scale = 2) val priceType: BigDecimal
)
